pkg/orchestrator/provider/aws/scanestimation/pricefetcher.go

```diff
@@ -0,0 +1,331 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scanestimation
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	ec2types "github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/aws/aws-sdk-go-v2/service/pricing"
+	"github.com/aws/aws-sdk-go-v2/service/pricing/types"
+
+	"github.com/openclarity/vmclarity/pkg/shared/utils"
+)
+
+type PriceFetcher interface {
+	GetSnapshotMonthlyCostPerGB(ctx context.Context, regionCode string) (float64, error)
+	GetVolumeMonthlyCostPerGB(ctx context.Context, regionCode string, volumeType ec2types.VolumeType) (float64, error)
+	GetDataTransferCostPerGB(sourceRegion, destRegion string) (float64, error)
+	GetInstancePerHourCost(ctx context.Context, regionCode string, instanceType ec2types.InstanceType, marketOption MarketOption) (float64, error)
+}
+
+type PriceFetcherImpl struct {
+	pricingClient *pricing.Client
+	ec2Client     *ec2.Client
+}
+
+// https://docs.aws.amazon.com/AmazonS3/latest/userguide/aws-usage-report-understand.html
+var regionCodeToUsageTypeAbbreviation = map[string]string{
+	"us-east-1":      "",     // N. virginia
+	"us-east-2":      "USE2", // Ohio
+	"us-west-1":      "USW1", // N. california
+	"us-west-2":      "USW2", // Oregon
+	"eu-central-1":   "EUC1", // Frankfurt
+	"eu-central-2":   "EUC2", // Zurich
+	"eu-south-1":     "EUS1", // Milan
+	"eu-west-1":      "EUW1", // Ireland
+	"eu-west-2":      "EUW2", // London
+	"eu-west-3":      "EUW3", // Paris
+	"af-south-1":     "CPT",  // Cape Town
+	"ap-east-1":      "APE1", // Hong Kong
+	"ap-southeast-3": "APS4", // Jakarta
+	"ap-south-2":     "APS5", // Hyderabad
+	"ap-southeast-4": "APS6", // Melbourne
+	"ca-central-1":   "CAN1", // Canada (Central)
+	"eu-north-1":     "EUN1", // Stockholm
+	"eu-south-2":     "EUS2", // Spain
+	"me-central-1":   "MEC1", // Middle East (UAE)
+	"me-south-1":     "MES1", // Middle East (Bahrain)
+	"sa-east-1":      "SAE1", // São Paulo
+	"us-gov-west-1":  "UGW1", // AWS GovCloud (US-West)
+	"us-gov-east-1":  "UGE1", // AWS GovCloud (US-East)
+	//"il-central-1": "EUS2", // Tel-Aviv // TODO couldn't find the abbreviation code
+
+}
+
+const (
+	usEast1RegionCode = "us-east-1"
+)
+
+func (o *PriceFetcherImpl) GetSnapshotMonthlyCostPerGB(ctx context.Context, regionCode string) (float64, error) {
+	var usageType string
+	if regionCode == usEast1RegionCode {
+		usageType = "EBS:SnapshotUsage"
+	} else {
+		abb, ok := regionCodeToUsageTypeAbbreviation[regionCode]
+		if !ok {
+			return 0, fmt.Errorf("failed to find usage type in regionCodeToUsageTypeAbbreviation map with regionCode %v", regionCode)
+		}
+		// example: USE2-EBS:SnapshotUsage
+		usageType = fmt.Sprintf("%v-EBS:SnapshotUsage", abb)
+	}
```

Comment:
maybe better to make this a helper function, `GetRegionalUsageType(regionCode, usageType) (string, error)` as we're doing it in multiple places.

Reply:
```
usageType, err = GetRegionUsageType("us-east-1", "EBS:SnapshotUsage")
if err != nil {
   return 0, err
}
fmt.Println(usageType)

EBS:SnapshotUsage
```

```
usageType, err = GetRegionUsageType("us-east-2", "EBS:SnapshotUsage")
if err != nil {
   return 0, err
}
fmt.Println(usageType)

USE2-EBS:SnapshotUsage
```

```
usageType, err = GetRegionUsageType("us-east-2", fmt.Sprintf("EBS:VolumeUsage.%v", volumeType))
if err != nil {
   return 0, err
}
fmt.Println(usageType)

USE2-EBS:VolumeUsage.spicy
```